pcap-mode.el

```diff
@@ -362,6 +374,32 @@ Argument FILTER-VALUE corresponds to the exact set of filters passed to `pcap-mo
   (if pcap-mode-reload-pcap-when-filter-changes
       (pcap-mode-reload-file)))
 
+(defun pcap-mode-set-named-filter (filter-name)
+  "Choose a predefined filter and apply it.
+Filters are obtained from the `pcap-mode-dfilters-file'."
+  (interactive (list (ido-completing-read
+		      "Filter name: "
+		      (if (file-readable-p pcap-mode-dfilters-file)
+			  (with-temp-buffer
+			    (insert-file-contents pcap-mode-dfilters-file)
+			    (goto-char (point-min))
+			    (cl-loop while (re-search-forward "^\"\\([^\"]*\\)\"" nil t)
+				     collect (match-string 1)))
+			(error "Can't read dfilters file: %s" pcap-mode-dfilters-file)))))
+  (let (filter)
```

Comment:
This code looks eerily similar to the interactive completion code.  Can you abstract this out into something like pcap-mode--dfilters-selection or something?